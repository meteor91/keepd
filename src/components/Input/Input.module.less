.input {
    font-size: 1rem;
    line-height: 1.7;
    font-weight: 400;
    color: #2D3643;
    background-color: #FFFFFF;
    border-radius: 6px;
    border: 1px solid #ced4da;
    padding: 11px 14px;
    width: 100%;

    &::placeholder {
        color: #AFBACA;
    }

    &:focus,
    &:focus-visible {
        outline: 0;
        border: 1px solid #5E718D;
    }

    &:disabled {
        border: 1px solid #D7DFE9;
    }

    &-error {
        color: #E92215;
        background: #FFF5F4;
        border: 1px solid #FFA19B;
    }
}