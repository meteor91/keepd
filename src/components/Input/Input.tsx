import React from "react";
import styles from './Input.module.less'

export interface InputProps {
    value: string;
    placeholder?: string;
    disabled?: boolean;
    error?: boolean;
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Input = (props : InputProps) => {
    const {
        value,
        placeholder,
        disabled,
        error,
        onChange
    } = props;
    return (
        <input
            className={`${styles.input} ${error ? styles['input-error']: ''}`}
            placeholder={placeholder}
            onChange={onChange}
            value={value}
            disabled={disabled}
        />
    )
}