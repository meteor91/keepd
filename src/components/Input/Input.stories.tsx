import React, { useState } from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import { Input, type InputProps } from './Input';

const meta: Meta<typeof Input> = {
    title: 'Input',
    component: Input,
};

export default meta;

type Story = StoryObj<typeof Input>;

type Props = Partial<InputProps>;

const InputPreview: React.FC<Props> = (props) => {
    const { value: initialValue, ...rest } = props;

    const [value, setValue] = useState(initialValue ?? '');

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
        setValue(event.target.value);
    };

    return (
        <Input
            onChange={handleChange}
            value={value}
            placeholder='placeholder'
            {...rest}
        />
    );
};

export const Basic: Story = {
    render: () => <InputPreview />,
    parameters: {
        variant: 'pocket',
    },
};

export const Disabled: Story = {
    render: () => <InputPreview disabled />,
    parameters: {
        variant: 'pocket',
    },
};

export const Error: Story = {
    render: () => <InputPreview error value='errorVal' />,
    parameters: {
        variant: 'pocket',
    },
};
