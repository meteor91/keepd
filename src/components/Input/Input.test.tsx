import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import {Input, InputProps} from './Input';

describe('Input', () => {
    const onChangeMock = jest.fn();

    const renderInput = (props: InputProps) => {
        return render(<Input {...props} />);
    };

    it('should render correct', () => {
        const {container} = renderInput({
            value: 'Test',
            placeholder: 'Enter text',
            error: true,
            onChange: onChangeMock,
        });

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should invoke onChange when change event fired', () => {
        renderInput({onChange: onChangeMock, value: 'test'});

        fireEvent.change(screen.getByRole('textbox'), {target: {value: 'hello'}});

        expect(onChangeMock).toHaveBeenCalledTimes(1);
        expect(onChangeMock).toHaveBeenCalledWith(expect.anything());
    });

    it('should be disabled when disabled prop is true', () => {
        renderInput({disabled: true, value: 'test'});

        expect(screen.getByRole('textbox')).toBeDisabled();
    });

    //
    // it('should render with error styles when error prop is true', () => {
    //   const { getByPlaceholderText, container } = renderInput({
    //     error: true,
    //     placeholder: 'Enter text',
    //     onChange: onChangeMock,
    //   });
    //   const inputElement = getByPlaceholderText('Enter text') as HTMLInputElement;
    //
    //   expect(inputElement.classList.contains('input-error')).toBe(true);
    //   expect(container.firstChild).toMatchSnapshot();
    // });


    // covered with snapshot
    // it('should render with the correct initial value', () => {
    //     const inputValue = 'Hello';
    //
    //     renderInput({value: inputValue, onChange: onChangeMock});
    //
    //     expect(screen.getByDisplayValue(inputValue)).toBeInTheDocument();
    // });
    //
    // it('should display placeholder', () => {
    //     const placeholder = 'Username';
    //
    //     renderInput({placeholder});
    //
    //     expect(screen.getByPlaceholderText(placeholder));
    // });
});
