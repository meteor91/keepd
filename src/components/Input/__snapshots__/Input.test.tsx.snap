// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correct 1`] = `
<input
  class="input input-error"
  placeholder="Enter text"
  value="Test"
/>
`;
